mlflow/pyfunc/model.py

```diff
@@ -175,9 +175,14 @@ def _save_model_with_class_artifacts_params(
         with TempDir() as tmp_artifacts_dir:
             tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
-            for artifact_name, artifact_uri in artifacts.items():
+            for artifact_index, (artifact_name, artifact_uri) in enumerate(artifacts.items()):
+                artifact_download_output_path = (
+                    os.path.join(tmp_artifacts_dir.path(), str(artifact_index)) if len(artifacts) > 1
```

Comment:
```suggestion
                    tmp_artifacts_dir.path(str(artifact_index)) if len(artifacts) > 1
```

Can we `str(artifact_index)` to `tmp_artifacts_dir.path`?